From 0b9c20f5e8a1d16f8995220ad78bb6ba07e425c2 Mon Sep 17 00:00:00 2001
From: ChonDoit <thphantomblog@gmail.com>
Date: Sun, 27 Oct 2024 21:52:42 +0100
Subject: [PATCH 2/7] Make status bar paddings overridables

with
persist.sys.phh.status_bar_padding_top
persist.sys.phh.status_bar_padding_start
persist.sys.phh.status_bar_padding_end
persist.sys.phh.status_bar_padding_bottom
---
 .../phone/KeyguardStatusBarView.java          |  9 +++--
 .../statusbar/phone/PhoneStatusBarView.java   | 34 +++++++++++++------
 2 files changed, 30 insertions(+), 13 deletions(-)

diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/phone/KeyguardStatusBarView.java b/packages/SystemUI/src/com/android/systemui/statusbar/phone/KeyguardStatusBarView.java
index 018124fdd..cd9cd80d1 100644
--- a/packages/SystemUI/src/com/android/systemui/statusbar/phone/KeyguardStatusBarView.java
+++ b/packages/SystemUI/src/com/android/systemui/statusbar/phone/KeyguardStatusBarView.java
@@ -27,6 +27,7 @@ import android.graphics.Color;
 import android.graphics.Insets;
 import android.graphics.Rect;
 import android.graphics.drawable.Drawable;
+import android.os.SystemProperties;
 import android.os.Trace;
 import android.util.AttributeSet;
 import android.util.TypedValue;
@@ -157,7 +158,7 @@ public class KeyguardStatusBarView extends RelativeLayout {
         // mStatusIconArea
         mStatusIconArea.setPaddingRelative(
                 mStatusIconArea.getPaddingStart(),
-                getResources().getDimensionPixelSize(R.dimen.status_bar_padding_top),
+                mStatusIconArea.getPaddingTop(),
                 mStatusIconArea.getPaddingEnd(),
                 mStatusIconArea.getPaddingBottom()
         );
@@ -209,8 +210,10 @@ public class KeyguardStatusBarView extends RelativeLayout {
         Resources res = getResources();
         mSystemIconsSwitcherHiddenExpandedMargin = res.getDimensionPixelSize(
                 R.dimen.system_icons_switcher_hidden_expanded_margin);
-        mStatusBarPaddingEnd = res.getDimensionPixelSize(
-                R.dimen.status_bar_padding_end);
+        mStatusBarPaddingEnd = SystemProperties.getInt("persist.sys.phh.status_bar_padding_end", -1);
+        if (mStatusBarPaddingEnd == -1)
+                mStatusBarPaddingEnd = res.getDimensionPixelSize(
+                        R.dimen.status_bar_padding_end);
         mMinDotWidth = res.getDimensionPixelSize(
                 R.dimen.ongoing_appops_dot_min_padding);
         mCutoutSideNudge = getResources().getDimensionPixelSize(
diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/phone/PhoneStatusBarView.java b/packages/SystemUI/src/com/android/systemui/statusbar/phone/PhoneStatusBarView.java
index 82f14393d..cb598d406 100644
--- a/packages/SystemUI/src/com/android/systemui/statusbar/phone/PhoneStatusBarView.java
+++ b/packages/SystemUI/src/com/android/systemui/statusbar/phone/PhoneStatusBarView.java
@@ -22,6 +22,7 @@ import android.content.res.Configuration;
 import android.content.res.Resources;
 import android.graphics.Insets;
 import android.graphics.Rect;
+import android.os.SystemProperties;
 import android.util.AttributeSet;
 import android.util.Log;
 import android.view.Display;
@@ -282,23 +283,36 @@ public class PhoneStatusBarView extends FrameLayout {
     }
 
     private void updatePaddings() {
-        int statusBarPaddingStart = getResources().getDimensionPixelSize(
-                R.dimen.status_bar_padding_start);
+        int statusBarPaddingTop = SystemProperties.getInt("persist.sys.phh.status_bar_padding_top", -1);
+        if(statusBarPaddingTop == -1)
+            statusBarPaddingTop = 0;
+
+        int statusBarPaddingStart = SystemProperties.getInt("persist.sys.phh.status_bar_padding_start", -1);
+        if(statusBarPaddingStart == -1)
+            statusBarPaddingStart = getResources().getDimensionPixelSize(R.dimen.status_bar_padding_start);
+
+        int statusBarPaddingEnd = SystemProperties.getInt("persist.sys.phh.status_bar_padding_end", -1);
+        if(statusBarPaddingEnd == -1)
+            statusBarPaddingEnd = getResources().getDimensionPixelSize(R.dimen.status_bar_padding_end);
+
+        int statusBarPaddingBottom = SystemProperties.getInt("persist.sys.phh.status_bar_padding_bottom", -1);
+        if(statusBarPaddingBottom == -1)
+            statusBarPaddingBottom = getResources().getDimensionPixelSize(R.dimen.status_bar_icons_padding_bottom);
 
         mStatusBarContents.setPaddingRelative(
-                statusBarPaddingStart,
-                getResources().getDimensionPixelSize(R.dimen.status_bar_padding_top),
-                getResources().getDimensionPixelSize(R.dimen.status_bar_padding_end),
-                0);
+            (int) statusBarPaddingStart,
+            (int) statusBarPaddingTop,
+            (int) statusBarPaddingEnd,
+            0);
 
         findViewById(R.id.notification_lights_out)
                 .setPaddingRelative(0, statusBarPaddingStart, 0, 0);
 
         findViewById(R.id.system_icons).setPaddingRelative(
-                getResources().getDimensionPixelSize(R.dimen.status_bar_icons_padding_start),
-                getResources().getDimensionPixelSize(R.dimen.status_bar_icons_padding_top),
-                getResources().getDimensionPixelSize(R.dimen.status_bar_icons_padding_end),
-                getResources().getDimensionPixelSize(R.dimen.status_bar_icons_padding_bottom)
+            (int) statusBarPaddingStart,
+            (int) statusBarPaddingTop,
+            (int) statusBarPaddingEnd,
+            (int) statusBarPaddingBottom
         );
     }
 
--
2.34.1
